src/Main.java

```diff
@@ -2,60 +2,52 @@ public class Main {
     public static void main(String[] args) {
 
         //task1
-        System.out.println("");
         char clientOS = 0;
         if (clientOS == 0) {
             System.out.println("Установите версию приложения для iOS по ссылке");
-        } else {
+        } else if (clientOS == 1) {
             System.out.println("Установите версию приложения для Android по ссылке");
         }
 
         //task2
-        System.out.println("");
         int clientDeviceYear = 2015;
-        if (clientDeviceYear >= 2015) {
-            if (clientOS == 0) {
+        if ((clientDeviceYear >= 2015) && (clientOS == 0)) {
                 System.out.println("Установите версию приложения для iOS по ссылке");
-            } else {
+        } else if ((clientDeviceYear >= 2015) && (clientOS == 1)) {
                 System.out.println("Установите версию приложения для Android по ссылке");
-            }
         } else if (clientOS == 0) {
             System.out.println("Установите облегченную версию приложения для iOS по ссылке");
-        } else {
+        } else if (clientOS == 1) {
             System.out.println("Установите облегченную версию приложения для Android по ссылке");
         }
 
         //task3
-        System.out.println("");
         int year = 2024;
-        int everyFourth = year % 4;
-        int everyHundredth = year % 100;
-        if (year > 1584) {
-            if ((everyFourth == 0) && (everyHundredth != 0)) {
-                System.out.println("Год високосный");
-            } else {
+        boolean everyFourth = (year % 4) == 0;
+        boolean everyHundredth = (year % 100) != 0;
+        boolean everyFourHundredth = (year % 400) == 0;
+        if ((year > 1584) && (everyFourth) && ((everyHundredth) || (everyFourHundredth))) {
```

Comment:
```suggestion
        boolean everyHundredth = year % 100 != 0;
        boolean everyFourHundredth = year % 400 == 0;
        if (year > 1584 && (everyFourth && everyHundredth || everyFourHundredth)) {
```
не нужно писать лишние скобки